studio/src/mas-repository.js

```diff
@@ -355,35 +401,89 @@ export class MasRepository extends LitElement {
+     * @returns {Promise<Object>} The saved fragment
      */
-    async saveFragment() {
-        Events.toast.emit({
-            variant: 'info',
-            content: 'Saving fragment...',
-        });
+    async saveFragment(fragment, options = {}) {
+        const { isInEditStore = true } = options;
+        const fragmentToSave = isInEditStore ? this.fragmentInEdit : fragment;
+        
+        if (!fragmentToSave) {
+            throw new Error('No fragment provided for saving');
+        }
+        
+        const isDictionaryFragment = fragmentToSave.path?.includes('/dictionary/');
+        this.showToast(isDictionaryFragment ? 'Saving placeholder...' : 'Saving fragment...');
         this.operation.set(OPERATIONS.SAVE);
+        
         try {
-            let updatedFragment = await this.aem.sites.cf.fragments.save(
-                this.fragmentInEdit,
+            let savedFragment;
+            const fragmentEndpoint = `${this.aem.cfFragmentsUrl}/${fragmentToSave.id}`;
+            
+            if (isDictionaryFragment) {
+                const response = await fetch(fragmentEndpoint, {
+                    method: 'GET',
+                    headers: {
+                        Accept: 'application/json',
+                        'Content-Type': 'application/json',
+                        ...this.aem.headers
+                    }
+                });
+                
+                if (!response.ok) {
+                    throw new Error(`Failed to get fragment for saving: ${response.status}`);
+                }
+                
+                const currentETag = response.headers.get('ETag');
+                
+                const { id, ...fragmentPayload } = fragmentToSave;
+                
+                const saveResponse = await fetch(fragmentEndpoint, {
+                    method: 'PUT',
+                    headers: {
+                        Accept: 'application/json',
+                        'Content-Type': 'application/json',
+                        'If-Match': currentETag,
+                        ...this.aem.headers
+                    },
+                    body: JSON.stringify(fragmentPayload)
+                });
+                
+                if (!saveResponse.ok) {
+                    throw new Error(`Failed to save fragment: ${saveResponse.status}`);
+                }
+                
+                savedFragment = await saveResponse.json();
+            } else {
+                savedFragment = await this.aem.sites.cf.fragments.save(fragmentToSave);
+                if (!savedFragment) throw new Error('Invalid fragment.');
+            }
+            
+            if (isInEditStore) {
+                this.fragmentStoreInEdit.refreshFrom(savedFragment);
+            }
+            
+            this.showToast(
+                isDictionaryFragment ? 'Placeholder successfully saved.' : 'Fragment successfully saved.',
+                'positive'
             );
```

Comment:
please move this all to a different "updatePlaceholderIndex" method you call from placeholder js together within createion code, and fix the payload, right now it's set to fragment id, hence the publish is broken atm

Reply:
done